tests/test_buffer.py

```diff
@@ -60,19 +70,78 @@ def test_line_single_sideded_right(self):
     def test_polygon(self):
         g = geometry.Polygon([[0, 0], [1, 0], [1, 1], [0, 1], [0, 0]])
         h = g.buffer(1, resolution=1)
-        assert h.geom_type == 'Polygon'
-        expected_coord = [(-1.0, 0.0), (-1.0, 1.0), (0.0, 2.0), (1.0, 2.0), (2.0, 1.0), (2.0, 0.0),
-                          (1.0, -1.0), (0.0, -1.0), (-1.0, 0.0)]
+        assert h.geom_type == "Polygon"
+        expected_coord = [
+            (-1.0, 0.0),
+            (-1.0, 1.0),
+            (0.0, 2.0),
+            (1.0, 2.0),
+            (2.0, 1.0),
+            (2.0, 0.0),
+            (1.0, -1.0),
+            (0.0, -1.0),
+            (-1.0, 0.0),
+        ]
         for index, coord in enumerate(h.exterior.coords):
             assert coord[0] == pytest.approx(expected_coord[index][0])
             assert coord[1] == pytest.approx(expected_coord[index][1])
 
     def test_polygon_single_sideded(self):
         g = geometry.Polygon([[0, 0], [1, 0], [1, 1], [0, 1], [0, 0]])
         h = g.buffer(1, resolution=1, single_sided=True)
-        assert h.geom_type == 'Polygon'
-        expected_coord = [(-1.0, 0.0), (-1.0, 1.0), (0.0, 2.0), (1.0, 2.0), (2.0, 1.0), (2.0, 0.0),
-                          (1.0, -1.0), (0.0, -1.0), (-1.0, 0.0)]
+        assert h.geom_type == "Polygon"
+        expected_coord = [
+            (-1.0, 0.0),
+            (-1.0, 1.0),
+            (0.0, 2.0),
+            (1.0, 2.0),
+            (2.0, 1.0),
+            (2.0, 0.0),
+            (1.0, -1.0),
+            (0.0, -1.0),
+            (-1.0, 0.0),
+        ]
         for index, coord in enumerate(h.exterior.coords):
             assert coord[0] == pytest.approx(expected_coord[index][0])
             assert coord[1] == pytest.approx(expected_coord[index][1])
+
+    def test_styles_values(self):
+        assert CAP_STYLE.round == 1
+        assert CAP_STYLE.round == BufferCapStyles.round
+        assert CAP_STYLE.flat == 2
+        assert CAP_STYLE.flat == BufferCapStyles.flat
+        assert CAP_STYLE.square == 3
+        assert CAP_STYLE.square == BufferCapStyles.square
+
+        assert JOIN_STYLE.round == 1
+        assert JOIN_STYLE.round == BufferJoinStyles.round
+        assert JOIN_STYLE.mitre == 2
+        assert JOIN_STYLE.mitre == BufferJoinStyles.mitre
+        assert JOIN_STYLE.bevel == 3
+        assert JOIN_STYLE.bevel == BufferJoinStyles.bevel
+
+    def test_styles(self):
+        g = geometry.LineString([[0, 0], [1, 0]])
+        h = g.buffer(1, cap_style=CAP_STYLE.round)
```

Comment:
I'd suggest flipping this, so that the expected value `h` is based on the currently recommended enum value, then the 2 assertions are based on the other forms (string, `CAP_STYLE.*`) to prove that they are equivalent to the recommended value.